substrate/frame/staking-async/src/pallet/impls.rs

```diff
@@ -880,6 +885,98 @@ impl<T: Config> Pallet<T> {
 	pub fn api_pending_rewards(era: EraIndex, account: T::AccountId) -> bool {
 		Eras::<T>::pending_rewards(era, &account)
 	}
+
+	/// Calculates the total stake of the lowest portion validators and stores it for the planned
+	/// era.
+	pub(crate) fn calculate_lowest_total_stake(era: EraIndex) {
+		// Only calculate if unbonding queue params have been set.
+		if let Some(params) = UnbondingQueueParams::<T>::get() {
+			// Determine the total stake from the lowest portion of validators and persist for the
+			// era.
+			let mut total_stake: Vec<_> =
+				ElectableStashes::<T>::get().into_iter().map(|(_, b)| b).collect();
+			let validators_to_check =
+				(params.lowest_ratio * total_stake.len() as u32).max(1) as usize;
+
+			// Sort exposure total stake by the lowest first and truncate to the lowest portion.
+			total_stake.sort();
+			total_stake.truncate(validators_to_check);
+
+			// Calculate the total stake of the lowest portion validators.
+			let total_stake = total_stake.into_iter().sum();
+
+			// Store the total stake of the lowest portion validators for the planned era.
+			Eras::<T>::set_lowest_stake(era, total_stake)
+		}
+	}
+
+	/// Curates the unlocking chunks for a stash account by:
+	/// - Calculating immediately withdrawable balance.
+	/// - Determining new unlock eras based on queue parameters.
+	/// - Organizing remaining chunks by era.
+	///
+	/// Returns a tuple of:
+	/// - Amount that can be immediately withdrawn.
+	/// - Map of era to remaining unlock chunks expected to be released in that era.
+	pub(crate) fn curate_unlocking_chunks(
+		stash: T::AccountId,
+	) -> ((EraIndex, BalanceOf<T>), BTreeMap<EraIndex, Vec<UnlockChunk<BalanceOf<T>>>>) {
+		let current_era = CurrentEra::<T>::get().unwrap_or(Zero::zero());
```

Comment:
As noted elsewhere, plasea read this from `struct Eras` for consitency.

Reply:
Refactored to use `Rotator::<T>::planned_era()`.